pymeasure/instruments/eurotest/eurotestHPP120256.py

```diff
@@ -0,0 +1,334 @@
+        LAM,OK Status OK""",
+        get_process=lambda v:
+        v[1].encode(EurotestHPP120256.response_encoding).decode('utf-8', 'ignore')
+    )
+
+    def emergency_off(self):
+        """ The output of the HV source will be switched OFF permanently and the values
+        of the voltage a current settings set to zero"""
+        log.info("Sending emergency off command to the instrument.")
+
+        self.write("EMCY OFF")
+        time.sleep(self.COMMAND_DELAY)
+
+    def shutdown(self, ramp):
+        """
+        Change the output voltage setting (V) to zero and
+        the ramp speed - changing rate (V/s) of the output voltage.
+        After calling shutdown, if the HV voltage output > 0
+        it should drop to zero at a certain rate given by the ramp parameter.
+        :param ramp: indicates the changing rate (V/s) of the voltage output
+        """
+        log.info(f"Executing the shutdown function with ramp: {ramp} V/s.")
+
+        self.ramp_to_zero(ramp)
+        super().shutdown()
+
+    def ramp_to_zero(self, ramp):
+        """
+        Sets the voltage output setting to zero and the ramp setting
+        to a determinated value done by the ramp parameter.
+        In summary, the method conducts (ramps) the voltage output to zero
+        at a determinated voltage changing rate (ramp in V/s).
+        :param ramp: Is the changing rate (ramp in V/s) for the ramp setting
+        """
+        log.info("Executing the ramp_to_zero function with ramp: {ramp} V/s.")
+
+        self.voltage_ramp = ramp
+        time.sleep(self.COMMAND_DELAY)
+        self.voltage = 0
+        time.sleep(self.COMMAND_DELAY)
+        self.voltage = 0
+        time.sleep(self.COMMAND_DELAY)
+
+    def wait_for_voltage_output_set(self, check_period=1.0, timeout=60.0):
+        """
+        Waits until HV voltage output reaches the voltage ouput setting.
+        Checks the voltage output every check_period seconds and raises an exception
+        if the voltage output doesn't reach the voltage setting until the timeout time.
+        :param check_period: voltage output will be measured every check_period (seconds) time
+        :param timeout: time (seconds) give to the voltage output to reach the voltage setting
+        :return: None
+        :raises: Exception if the voltage output can't reach the voltage setting
+        before the timeout completes (seconds)
+        """
+        log.info("Executing the wait_for_voltage_output_set function.")
+
+        ref_time = time.time()
+        future_time = ref_time + timeout
+
+        voltage_output_setting = self.voltage
+        time.sleep(self.COMMAND_DELAY)
+        error = 0.02  # error of +-ten volts (0.01kV) to enter into the voltage stable window
+        voltage_output = self.measure_voltage
+        voltage_output_set = (voltage_output > (voltage_output_setting - error)) and \
+                             (voltage_output < (voltage_output_setting + error))
```

Comment:
```suggestion
        error = 0.02  # error of +-ten volts (0.01kV) to enter into the voltage stable window
        voltage_output = self.measure_voltage
        voltage_output_set = math.isclose(voltage_output, voltage_output_setting, reltol=0.0, atol=error)
```
Same below, use Python's batteries :-)
`voltage_output_reached` might be a clearer variable name.

Reply:
Perfect!!! I didn't know that math function. Thanks